components/src/PurchaseOrderWidget/purchase.order.widget.tsx

```diff
@@ -0,0 +1,194 @@
+    this.startValidationTimer = this.startValidationTimer.bind(this);
+    this.clearIsTypingTimer = this.clearIsTypingTimer.bind(this);
+    this.verifyPONumber = this.verifyPONumber.bind(this);
+    this.renderInputStatus = this.renderInputStatus.bind(this);
+    this.updateInputState = this.updateInputState.bind(this);
+    this.showMorePODetails = this.showMorePODetails.bind(this);
+  }
+
+  clearIsTypingTimer(event) {
+    if (isTypingTimer != null) {
+      this.setState({ inputStatus: POInputStates.LOADING });
+      clearTimeout(isTypingTimer);
+      isTypingTimer = null;
+    }
+  }
+
+  updateInputState(event) {
+    this.setState({ inputTextValue: event.target.value });
+  }
+
+  verifyPONumber() {
+    const { inputTextValue } = this.state;
+
+    // NOTE: This should be replaced with a network request to validate the PO number entered.  A timeout is placed here as a placeholder.
+    setTimeout(() => {
+      if (dummyValidPONumbers.has(inputTextValue)) {
+        this.setState({ inputStatus: POInputStates.VERIFIED });
+      } else {
+        this.setState({ inputStatus: POInputStates.ERROR });
+      }
+    }, 1000);
+
+    isTypingTimer = null;
+  }
+
+  startValidationTimer(event) {
+    const { timeoutBeforeVerify } = this.props;
+    if (isTypingTimer == null) {
+      const timeID = setTimeout(() => {
+        this.verifyPONumber();
+      }, timeoutBeforeVerify);
+      this.setState({ inputStatus: POInputStates.LOADING });
+      isTypingTimer = timeID;
+    }
+  }
+
+  renderInputStatus() {
+    const { inputStatus } = this.state;
+
+    switch (inputStatus) {
+      case POInputStates.LOADING:
+        return (
+          <div className="inputLoader-container">
+            <div className="inputLoader" />
+          </div>);
+      case POInputStates.VERIFIED:
+        return (
+          <div className="inputLoader-container">
+            <div className="checkmark chosen" />
+          </div>);
+      default:
+        return null;
+    }
+  }
+
+  renderPurchaseOrderTextBox() {
+    const { inputTextValue, inputStatus } = this.state;
+    const warningMessage = {
+      debugMessages: 'Invalid PO number',
+      type: 'error',
+      id: 'field.invalid.email.format',
+    };
+    return (
+      <div>
+        <div className="purchase-order-widget-input-container">
+          <input value={inputTextValue} className="form-control" type="text" placeholder="Enter Purchase Order Number (PO)" onChange={this.updateInputState} onKeyUp={this.startValidationTimer} onKeyDown={this.clearIsTypingTimer} />
+          {this.renderInputStatus()}
+        </div>
+        {
+          inputStatus === POInputStates.ERROR && (
+            <div>
+              <Messagecontainer message={[warningMessage]} />
+            </div>
+          )
+        }
+      </div>
+    );
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  showMorePODetails() {
+    // TODO: Need to implement function that shows the PO details modal.
```

Comment:
Is this something we need to implement, or should be implemented by the dev that consumes this?